### ISSUE TITLE
Range by - return any period it enters, not just fully elapsed periods

### ISSUE THREAD

**Issue by ljubic** [rotaready/moment-range#198]
```javascript
var startDate = moment("2017-11-30");
var endDate = moment("2018-03-30");

var range =moment.range(startDate, endDate);
```

```javascript
Array.from(range.by('years')).map(e => e.format('YYYY'))
```
Opposite of your example on the homepage it returns only ["2017"] although it should return 2018 as well. It works well with range by months or days.

**Comment by TristanJM** [comment]
Hey, thanks for the feedback.

This **is intended** behaviour as it only adds the an item to the iterator if the entire period has elapsed.

In your case, an entire year has not passed in the range so it only takes the 2017 from the start of the range.
If your end date was 2017-11-30, you would then then get `["2017","2018"]`.

I will add a new option to also return any period it enters in the next release.

**Comment by ljubic** [comment]
Ok. It is solvable by creating range array by parsing years from dates if somebody needs it:
```javascript
//exclusive end
const rangeArray = (start, end) => Array.from({length: (end - start)}, (v, k) => k + start);

let years = rangeArray(moment("2017-11-30").year(), moment("2018-03-30").year() + 1);
```

**Comment by 92kz1h** [comment]
@TristanJM Was this included in 3.1.0?

**Comment by TristanJM** [comment]
No, we prioritised some other features.
Will add to 3.2.0 milestone.

Does the above solution work for you?

**Comment by 92kz1h** [comment]
Thanks for the rapid response. Essentially it works, slightly different:

```
range = moment.range(fromDate.startOf('month'), toDate.endOf('month'))
```

where `fromDate` and `toDate` are already Moment objects.

**Comment by gf3** [comment]
what about something like a `snapTo()` method that essentially does the above? e.g.

``` js
const start = moment("2017-01-01T13:30:00");
const end = moment("2017-01-05T01:45:12");
const range = moment.range(start, end);

Array.from(range.by('days')).map(m => m.format('YY-MM-DD')) // returns ↓
[ '17-01-01', '17-01-02', '17-01-03', '17-01-04' ]

range.snapTo('day');
range.start // moment("2017-01-01T00:00:00")
range.end // moment("2017-01-05T23:59:59")

Array.from(range.by('days')).map(m => m.format('YY-MM-DD')) // returns ↓
[ '17-01-01', '17-01-02', '17-01-03', '17-01-04', '17-01-05' ]
```

**Comment by gf3** [comment]
Closed by #220
